$NetBSD: patch-ad,v 1.5 2011/12/19 21:36:07 ryoon Exp $

* "Should be one line" test is for gcc 4.5's cpp from emacs20's patch-ab.

--- configure.orig	1995-06-05 01:11:38.000000000 +0000
+++ configure
@@ -581,12 +581,14 @@ case "${canonical}" in
   *-*-netbsd* )
     opsys=netbsd
     case "${canonical}" in
+      alpha-*-netbsd*)	machine=alpha ;;
       i[345]86-*-netbsd*) machine=intel386 ;;
       m68k-*-netbsd*)
 			# This is somewhat bogus.
 			machine=hp9000s300 ;;
       mips-*-netbsd*)	machine=pmax ;;
       ns32k-*-netbsd*)	machine=ns32000 ;;
+      powerpc-*-netbsd*) machine=powerpc ;;
       sparc-*-netbsd*)	machine=sparc ;;
     esac
   ;;
@@ -1177,6 +1179,7 @@ case "${canonical}" in
       *-bsd386* | *-bsdi* )	opsys=bsd386 ;;
       *-386bsd* )	        opsys=386bsd ;;
       *-freebsd* )	        opsys=freebsd ;;
+      *-dragonfly* )		opsys=dragonfly ;;
       *-nextstep* )             opsys=mach2 ;;
       ## Otherwise, we'll fall through to the generic opsys code at the bottom.
     esac
@@ -1384,10 +1387,13 @@ if test -z "$CPP"; then
 #include "confdefs.h"
 #include <stdio.h>
 Syntax Error
+Should \\
+be one line.
 EOF
 # Some shells (Coherent) do redirections in the wrong order, so need
 # the parens.
-ac_err=`eval "($ac_cpp conftest.${ac_ext} >/dev/null) 2>&1"`
+ac_err=`eval "($ac_cpp conftest.${ac_ext} > conftest2.out) 2>&1 ; \
+grep -L \"Should  *be one line\" conftest2.out"`
 if test -z "$ac_err"; then
   :
 else
@@ -1397,10 +1403,13 @@ else
 #include "confdefs.h"
 #include <stdio.h>
 Syntax Error
+Should \\
+be one line.
 EOF
 # Some shells (Coherent) do redirections in the wrong order, so need
 # the parens.
-ac_err=`eval "($ac_cpp conftest.${ac_ext} >/dev/null) 2>&1"`
+ac_err=`eval "($ac_cpp conftest.${ac_ext} >/dev/null) 2>&1 ; \
+grep -L \"Should  *be one line\" conftest2.out"`
 if test -z "$ac_err"; then
   :
 else
